internals/approval_defs.py

```diff
@@ -161,8 +162,21 @@ def is_resolved(approval_values, field_id):
   return False
 
 def set_vote(
-    feature_id: int, gate_id: int, new_state: int, set_by_email: str) -> None:
+    feature_id: int,  gate_type: int, new_state: int, set_by_email: str,
+    gate_id: Optional[int]=None) -> None:
   """Add or update an approval value."""
+  gate: Optional[Gate] = None
+  if not gate_id:
```

Comment:
This would be more clear if you reversed the condition and swapped the then and else branches.

Reply:
This went through some iterations and some style got lost in translation here - I've updated this line to
```
if gate_id is None:
  ...
```
which is hopefully more legible.